application/controllers/Hook.php

```diff
@@ -0,0 +1,254 @@
+<?php
+defined('BASEPATH') OR exit('No direct script access allowed');
+
+class Hook extends MY_Base {
+
+    private $execution_id = null;
+    private $pattern = '/[0-9]{4}-[0-9]{2}-[0-9]{2}-(.*)?\.json/';
+    private $force = false;
+
+    public function add()
+    {
+        $this->load->model('Execution');
+        $this->load->model('Suite');
+        $this->load->model('Test');
+
+        log_message('info', "verifying data");
+        if (!$this->input->get('token') || !$this->input->get('filename')) {
+            header($_SERVER['SERVER_PROTOCOL'] . ' 400 Bad Request', true, 400);
+            exit("no enough parameters");
+        }
+
+        log_message('info', "verifying name of file (".$this->input->get('filename').")");
+        preg_match($this->pattern, $this->input->get('filename'), $matches);
+        if (!isset($matches[1])) {
+            header($_SERVER['SERVER_PROTOCOL'] . ' 400 Bad Request', true, 400);
+            exit("filename '".$this->input->get('filename')."' is invalid");
+        }
+
+        log_message('info', "verifying token is here");
+        if (!$this->checkToken($this->input->get('token'))) {
+            header($_SERVER['SERVER_PROTOCOL'] . ' 400 Bad Request', true, 400);
+            exit("invalid token");
+        }
+
+        if ($this->input->get('force') !== NULL) {
+            $this->force = true;
+        }
+
+        //get the file from the GCP API
+        $filename = $this->input->get('filename');
+        log_message('info', "receiving filename $filename");
+        //create URL
+        $url = sprintf("https://storage.googleapis.com/prestashop-core-nightly/reports/%s", $filename);
```

Comment:
Would it be wise to variablize this URL/Storage ?

Reply:
That's a good suggestion. I thought about it but i'm not sure to know where is the best place to put it ? It's a little too big to be in an environment variable IMHO

Reply:
It's really not that big :)
And, if possible, I'd rather have it as a configuration parameter.

Reply:
OK, I put it as a private property of that class, is that good enough ? Or would you prefer it as an environment variable so that we can change it through the Docker script

Reply:
Environment variable is always nice 👍

Reply:
You can also provide a default value and allow it to be override to make config simpler 😄

Reply:
I don't think it's something that will change anytime soon, so I prefer to keep it as a private property for now.